<launch>
	<arg name="rate" default="10" />
	<arg name="use_nodelets" default="false" />
	<arg name="nodelet_manager_name" default="nodelet_manager" />
	<arg name="initialize" default="false" />
	<arg name="param_file_name" default="test_params" />
	<arg name="name_prefix" default="color_classifier" />
	<arg name="name_postfix" default="" />
	<arg name="image" default="/image_server/output_image" />
	<arg name="image_transport" default="raw" />
	
	<arg name="name" value="$(arg name_prefix)$(arg name_postfix)" />
	<arg name="pkg" value="color_classifier" />
	<arg name="node_type" value="$(arg pkg)_node" />
	<arg name="nodelet_type" value="$(arg pkg)_nodelet" />

	<group unless="$(arg use_nodelets)" >
		<node pkg="$(arg pkg)" type="$(arg node_type)" name="$(arg name)" output="screen" args="_rate:=$(arg rate) ~image:=$(arg image) _image_transport:=$(arg image_transport)" >
			<rosparam unless="$(arg initialize)" param="reconfigure" command="load" file="$(find color_classifier)/params/$(arg param_file_name).yaml" />
		</node>
	</group>

	<group if="$(arg use_nodelets)" >
		<node pkg="nodelet" type="nodelet" name="$(arg name)" output="screen" args="load $(arg pkg)/$(arg nodelet_type) $(arg nodelet_manager_name) _rate:=$(arg rate) ~image:=$(arg image) _image_transport:=$(arg image_transport)" >
			<rosparam unless="$(arg initialize)" param="reconfigure" command="load" file="$(find color_classifier)/params/$(arg param_file_name).yaml" />
		</node>
	</group>

	<arg name="show_reconfigure" default="true" />
	<include file="$(find seabee3_launch)/dynamic_reconfigure.launch" >
		<arg name="enable" value="$(arg show_reconfigure)" />
		<arg name="name" value="$(arg name)" />
	</include>
	
	<arg name="show_image" default="true" />
	<include file="$(find seabee3_launch)/image_view.launch" >
		<arg name="enable" value="$(arg show_image)" />
		<arg name="name" value="$(arg name)" />
		<arg name="image_transport" value="$(arg image_transport)" />
	</include>
</launch>
